### PR TITLE
fix: cp sdk clone inParams&globalStates

### PR DESCRIPTION
**Pull request by sfwn**
#### What this PR does / why we need it:

fix: cp sdk clone inParams&globalStates

#### Specified Reivewers:

/assign @Effet

### CODE REVIEW

File: providers/component-protocol/cpregister/register_comp_provider.go [hunk 1]
```diff
@@ -53,7 +55,12 @@ func RegisterProviderComponent(scenario, componentName string, providerPtr Compo
 	providerName := cputil.MakeComponentProviderName(scenario, componentName)
 
 	// register component
-	RegisterComponent(scenario, componentName, func() cptype.IComponent { return providerPtr })
+	RegisterComponent(scenario, componentName, func() cptype.IComponent {
```
Comment (Effet) [inline providers/component-protocol/cpregister/register_comp_provider.go:58]: separate a func like `wrapXxx`, and add some ut

File: providers/component-protocol/cptype/sdk.go [hunk 2]
```diff
@@ -37,19 +38,42 @@ type SDK struct {
...
+	// inParams
+	clonedInParams := make(InParams)
+	for k, v := range sdk.InParams {
+		clonedInParams[k] = v
+	}
+	clonedSDK.InParams = clonedInParams
+	// globalStates
+	clonedGlobalStates := make(GlobalStateData)
+	for k, v := range *sdk.GlobalState {
+		clonedGlobalStates[k] = v
```
Comment (Effet) [inline providers/component-protocol/cptype/sdk.go:72]: what if `v` is a pointer?

**Review by Effet — CHANGES_REQUESTED** [review]

File: providers/component-protocol/cpregister/register_comp_provider.go (hunk 1 above)
Comment (sfwn) [reply providers/component-protocol/cpregister/register_comp_provider.go:58]: adjusted

File: providers/component-protocol/cptype/sdk.go (hunk 2 above)
Comment (sfwn) [reply providers/component-protocol/cptype/sdk.go:72]: No problem here.
It's a range pitfall, if you don't save address of `v`, it's ok.

File: providers/component-protocol/cptype/sdk.go [hunk 3]
```diff
@@ -37,19 +38,42 @@ type SDK struct {
 	Event       ComponentEvent
 	CompOpFuncs map[OperationKey]OperationFunc
 	Comp        *Component
+
+	// for parallel use, it's request level
+	StdStructuredPtr IStdStructuredPtr
+
+	Lock       sync.Mutex
+	OriginLock *sync.Mutex
```
Comment (Effet) [inline providers/component-protocol/cptype/sdk.go:46]: should not put Origin SDK's Lock into new SDK, i suggest you can define an OriginSDK aside with SDK

**Review by Effet — CHANGES_REQUESTED** [review]

**Review by Effet — APPROVED** [review]

File: providers/component-protocol/cptype/sdk.go (hunk 3 above)
Comment (sfwn) [reply providers/component-protocol/cptype/sdk.go]: adjusted

**Pull request merged**